### PR TITLE
Nam 35

### PR DESCRIPTION
**Pull request by BruceMcl**
Creating PR for NAM-35 so that markerhistory can merge into dev. And can hopefully be linked into the front end for changing markers.

### CODE REVIEW

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 1]
```diff
@@ -61,6 +63,7 @@
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.exception.ExceptionUtils;
 import org.apache.commons.lang3.math.NumberUtils;
+import org.hibernate.Session;
```
Comment (gillmanjc) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: Is this import used in the file? Don't see it being used in the method that was added

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 2]
```diff
@@ -4060,4 +4067,35 @@ private String getReviewError(Long status){
...
+		try {
+			assignment = getAssignment(assignmentId);
+		
+    	String context = assignment.getContext();
+    	double oldQuota = newAssignmentMarker.getQuotaPercentage();
+    	double newQuota = oldQuota + newAssignmentMarker.getQuotaPercentage();
+    	
+    	String modifier =  userDirectoryService.getCurrentUser().getId();
+    	java.util.Date uDate = new java.util.Date();
+    	java.sql.Date modifiedDate = new java.sql.Date(uDate.getTime());
```
Comment (gillmanjc) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: Why are these date objects created if not used?

File: assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentService.java [hunk 3]
```diff
@@ -28,6 +28,7 @@
 import org.sakaiproject.assignment.api.model.Assignment;
 import org.sakaiproject.assignment.api.model.AssignmentSubmission;
 import org.sakaiproject.assignment.api.model.AssignmentSubmissionSubmitter;
+import org.sakaiproject.assignment.api.model.AssignmentMarker;
```
Comment (gillmanjc) [inline assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentService.java]: Is this needed? Don't see any use for the import in the file?

File: assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentService.java [hunk 4]
```diff
@@ -737,5 +738,5 @@ public String getDeepLinkWithPermissions(String context, String assignmentId, bo
 
     String getUsersLocalDateTimeString(Instant date);
 
-    public List<ContentReviewResult> getContentReviewResults(AssignmentSubmission submission);
+    public List<ContentReviewResult> getContentReviewResults(AssignmentSubmission submission);
```
Comment (gillmanjc) [inline assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentService.java]: Just a space change?

File: assignment/api/src/java/org/sakaiproject/assignment/api/persistence/AssignmentRepository.java [hunk 5]
```diff
@@ -80,4 +81,12 @@ void newSubmission(Assignment assignment,
     long countAssignmentSubmissions(String assignmentId, Boolean graded, Boolean hasSubmissionDate, Boolean userSubmission);
 
     void resetAssignment(Assignment assignment);
+
+    
+    /**
+     * NAM-35
+     */
+    
+	void logMarkerChanges(Assignment assignment, AssignmentMarker oldAssignmentMarker,
```
Comment (gillmanjc) [inline assignment/api/src/java/org/sakaiproject/assignment/api/persistence/AssignmentRepository.java]: Please change method signature to take an "AssignmentMarker" object as parameter and not a list of parameters.. see  newAssignment(Assignment assignment) as example

Comment (BruceMcl) [reply assignment/api/src/java/org/sakaiproject/assignment/api/persistence/AssignmentRepository.java]: I used newSubmission as the example here, as we needed the assignment and the AssignmentMarkers (old and new). The context, quotes and modifier we should be able to derive in processing. I can remove them from the parameter list.  Would this be alright?

Comment (gillmanjc) [reply assignment/api/src/java/org/sakaiproject/assignment/api/persistence/AssignmentRepository.java]: I thought that were ever you call logMarkerChanges from, those values must then be populated on an AssignmentMarkerHistory object, and then the method signature can just look something like this logMarkerChanges(AssignmentMarkerHistory assignmentMarkerHistory ), but I think its fine so will resolve.

**Comment by gillmanjc** [comment]
Please go ahead with merge.

**Pull request merged**